source/reflect.h2

```diff
@@ -783,6 +793,21 @@ auto parser::apply_type_meta_functions( declaration_node& n )
             return false;
         }
     }
+
+    // Ensure the initializers of members lowered as bases
+    // are used by injecting the default constructor here.
+    if (auto res = n.should_add_default_constructor(); res.value) {
+        if (!n.member_function_generation) {
+            error(
+                "'@struct' scope object can have an initializer if it is ordered after all 'this' objects only",
```

Comment:
This `apply_type_meta_functions` function should only be dispatching the metafunction names to apply them, rather than do additional logic / side effects...

If this check is `struct`-specific, should it go inside `cpp2_struct`?

Otherwise, if it's not `struct`-specific, should it be in `cppfront.cpp`, maybe some where around [here](https://github.com/hsutter/cppfront/blob/main/source/cppfront.cpp#L5216-L5220)?

Reply:
I'll see what I can do.

I did it here because a metafunction is so much simpler.
The `add_default_constructor` metafunction adds `operator=: (out this) = { }`,
and so `cppfront.cpp` generates a default constructor that considers the member initializers.

Generation can't happen with a `struct`.
To not ignore their initializers, the check is needed.

Reply:
I think the best would be some refactoring
to move this piece of code after the call to `apply_type_meta_functions`.

The less I have to manually specify parse states, the better.
In that case, moving it to `cppfront.cpp` is the worst.
That has me putting a lot of effort to manually specify a `declaration_node` structure
to mimic the fraction of power it takes for `add_default_constructor(t)` to do that.

The alternative is manual generation of Cpp1 to attempt to achieve the same.

Reply:
Applied with commit 386219da3aa1e9fa7c740c5ea48b4fc08c1dc4dd.